lib/adaptivequemgr.go

```diff
@@ -138,11 +138,12 @@ func bindEvictNameOk(bindName string) (bool) {
 bind name and values */
 func (mgr *adaptiveQueueManager) doBindEviction() (int) {
```

Comment:
The outer lock on bindevict is being held for too long. It gets unlocked only after inner lock on workerpool is done processing dispatched workers.